### DISCUSSION TITLE
fq_codel supported but

### DISCUSSION THREAD

**Discussion by dtaht** [accel-ppp/accel-ppp#151] [Shaper]
It's very cool to see you've added fq_codel support. However in looking over your API, there are some things that could use tweaking. 

1) the fq_codel "target" parameter needs to be scaled to 1 MTU, so at 1mbit that's target 15ms. 

2) the fq_codel quantum parameter is generally better set to 300 

There's some other potentially nagging API details, I recently tried to write some up over on mikrotiks site here: https://forum.mikrotik.com/viewtopic.php?t=179307 - and in some cases we are seeing a non-standardized dhcp message that can be sent to clients to configure their shaper if available, which I'd like to see developed further, as it's often better to shape the egress from the actual egress....

4) cake is the new hotness, and has an integral shaper among other things, and handling the mtu problem above. See how it's used in LibreQos, also.  It's been in linux since 4.19. 

5) It's kind of unclear to me how each of these virtual interfaces are setup from a quick look at the code. Is there one per customer?

**Comment by svlobanov** [comment]
5. yes, in general, accel-ppp creates or use pre-created interface for each customer
1,2. where can I read about "better" params for quantum and target? can you provide a link to authoritative source where it is explained (like research paper) ?

I didn't get an idea related to dhcp&shaper. can you please explain it more detailed?

**Reply by dtaht** [reply]
Cake?
